dev-packages/node-integration-tests/utils/runner.ts

```diff
@@ -164,6 +165,69 @@ type StartResult = {
   ): Promise<T | undefined>;
 };
 
+export function createEsmAndCjsTests(
+  cwd: string,
+  scenarioPath: string,
+  instrumentPath: string,
+  callback: (
+    createTestRunner: () => ReturnType<typeof createRunner>,
+    testFn: typeof test | typeof test.fails,
+    mode: 'esm' | 'cjs',
+  ) => void,
+  options?: { failsOnCjs?: boolean; failsOnEsm?: boolean },
+): void {
+  const mjsScenarioPath = join(cwd, scenarioPath);
+  const mjsInstrumentPath = join(cwd, instrumentPath);
+
+  if (!mjsScenarioPath.endsWith('.mjs')) {
+    throw new Error(`Scenario path must end with .mjs: ${scenarioPath}`);
+  }
+
+  if (!existsSync(mjsInstrumentPath)) {
+    throw new Error(`Instrument file not found: ${mjsInstrumentPath}`);
+  }
+
+  const cjsScenarioPath = join(cwd, `tmp_${scenarioPath.replace('.mjs', '.cjs')}`);
+  const cjsInstrumentPath = join(cwd, `tmp_${instrumentPath.replace('.mjs', '.cjs')}`);
+
+  // For the CJS runner, we create some temporary files...
+  if (!options?.failsOnCjs) {
```

Comment:
l: if `failsOnCjs` is `true` and we don't convert the files, would we even run the test where we add `test.fails` further down below?

Reply:
good catch, forgot about this - I restructured this a few times but forgot about this, this should always be run :)